### ISSUE TITLE
resolvelib.resolvers.ResolutionImpossible: dask, pyct, rasterio

### ISSUE THREAD

**Issue by InLaw** [DavHau/mach-nix#62]
Hello 

i try to use mach-nix 2.2.0

before I used poetry to resulve the pkgs and export the resolution to a file (which I try to feed in mach-nix)

but I get:

```
resolvelib.resolvers.ResolutionImpossible: [RequirementInformation(requirement=Requirement.parse('dask==2.22.0'), parent=None), RequirementInformation(requirement=Requirement.parse('dask[complete]>=0.18.0'), parent=Candidate(name='datashader', ver=<Version('0.10.0')>, extras=()))]
builder for '/nix/store/d5dnw42d7jfhdqmzh35cybak8w91jpkh-mach_nix_file.drv' failed with exit code 1
error: --- Error --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix-build
build of '/nix/store/d5dnw42d7jfhdqmzh35cybak8w91jpkh-mach_nix_file.drv' failed
```

`resolvelib.resolvers.ResolutionImpossible: [RequirementInformation(requirement=Requirement.parse('pyct==0.4.6'), parent=None), RequirementInformation(requirement=Requirement.parse('pyct[cmd]'), parent=Candidate(name='datashader', ver=<Version('0.10.0')>, extras=()))]`

`resolvelib.resolvers.ResolutionImpossible: [RequirementInformation(requirement=Requirement.parse('rasterio==1.1.5'), parent=None), RequirementInformation(requirement=Requirement.parse('rasterio[s3]>=1.0.28'), parent=Candidate(name='rio-cogeo', ver=<Version('1.1.10')>, extras=()))]`

and via nix-shell shll.nix execution:
`
KeyError: 'numba'
resolvelib.resolvers.ResolutionImpossible: [RequirementInformation(requirement=Requirement.parse('numba==0.50.1'), parent=None)]`

`KeyError: 'arcgis'
`

---
if I skip those I get 

`
RecursionError: maximum recursion depth exceeded while getting the str of an object
`

---

via nix-shell
and mach-nix

---

```
let
  mach-nix = import (builtins.fetchGit {
    url = "https://github.com/DavHau/mach-nix/";
    ref = "2.2.0";
  });
in mach-nix.mkPython {
  requirements = ''
     # your requirements here
     dask==2.22.0
     pyct==0.4.6
     rasterio==1.1.5
     numba==0.50.1
     arcgis==1.8.2
  '';
}
```

**Comment by DavHau** [comment]
Thanks for the report. You seem to hit multiple issues here.
One of them might be the poetry export itself. see https://github.com/python-poetry/poetry/issues/1970
But you're definitely also hitting a bug in mach-nix here.
Could you please provide an complete nix expression in the following form which causes the problem.
```nix
let
  mach-nix = import (builtins.fetchGit {
    url = "https://github.com/DavHau/mach-nix/";
    ref = "2.2.0";
  });
in mach-nix.mkPython {
  requirements = ''
     # your requirements here
  '';
}
```
Otherwise it's very difficult for me to reproduce the issue. Thanks!

**Comment by InLaw** [comment]
> Thanks for the report. You seem to hit multiple issues here.
> One of them might be the poetry export itself. see [python-poetry/poetry#1970](https://github.com/python-poetry/poetry/issues/1970)
> But you're definitely also hitting a bug in mach-nix here.
> Could you please provide an complete nix expression in the following form which causes the problem.
> 
> ```nix
> let
>   mach-nix = import (builtins.fetchGit {
>     url = "https://github.com/DavHau/mach-nix/";
>     ref = "2.2.0";
>   });
> in mach-nix.mkPython {
>   requirements = ''
>      # your requirements here
>   '';
> }
> ```
> 
> Otherwise it's very difficult for me to reproduce the issue. Thanks!

**Comment by DavHau** [comment]
The python 3.8 wheel finding bugfix in version 2.2.1 fixes the problem caused by `numba==0.50.1`.

Also the problems with `dask[complete]` and `datashader` should be fixed. Those packages have a circular dependency on each other. Mach-nix is now able to handle this since 2.2.1
